### ISSUE TITLE
Admin 'Adapters' page does not show correct version info

### ISSUE THREAD

**Issue by raintonr** [ioBroker/ioBroker.admin#548]
The 'Adapters' page does not show the actual versions installed. Eg:

![image](https://user-images.githubusercontent.com/3794098/79629735-3252b480-814c-11ea-8ec0-83a609e0360c.png)

Compare this to the actual versions as reported on the command line:

```
# iobroker update | grep installed
Adapter    "admin"         : 4.0.7    , installed 4.0.7
Adapter    "backitup"      : 1.4.5    , installed 1.4.5
Adapter    "chromecast"    : 2.3.1    , installed 2.3.1
Adapter    "discovery"     : 2.3.1    , installed 2.3.1
Adapter    "influxdb"      : 1.7.0    , installed 1.7.0
Adapter    "info"          : 1.7.0    , installed 1.7.0
Adapter    "javascript"    : 4.5.1    , installed 4.5.1
Controller "js-controller" : 2.2.9    , installed 2.2.9
Adapter    "loxone"        : 1.0.0    , installed 1.0.0
Adapter    "sayit"         : 1.8.2    , installed 1.8.2
Adapter    "simple-api"    : 2.4.0    , installed 2.4.0
Adapter    "socketio"      : 3.0.3    , installed 3.0.3
Adapter    "sony-bravia"   : 1.0.1    , installed 1.0.1
Adapter    "weatherunderground": 3.2.1   , installed 3.2.1
Adapter    "web"           : 3.0.4    , installed 3.0.4
Adapter    "zwave"         : 2.0.0    , installed 2.0.0
```
FYI...

```
# iobroker -v
2.2.9
```

**Comment by Apollon77** [comment]
ALso after updating the page (The Arrow button top left)

**Comment by raintonr** [comment]
Yes. I tried hitting the 'update adapter information' button several times.

**Comment by GermanBluefox** [comment]
Could `iobroker upload admin` help?

**Comment by Apollon77** [comment]
Please also try deleting browser cache and local storage.

**Comment by raintonr** [comment]
```
# iob upload admin
upload [3] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/words.js words.js application/javascript
upload [2] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index_m.html index_m.html text/html
upload [1] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index.html index.html text/html
upload [0] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/admin.png admin.png image/png
```

That didn't help. I reload the admin page and refresh but adapter details still the same (ie. incorrect).

**Comment by Apollon77** [comment]
Please install Admin 4.0.8 !! 4.0.7 is broken in one area

**Comment by raintonr** [comment]
OK.... what is going on here? I run this on the command line:

```
# iobroker upgrade admin
Update admin from @4.0.7 to @4.0.8
host.pl.mh Adapter "system.adapter.admin.0" is stopped.
NPM version: 6.13.4
npm install iobroker.admin@4.0.8 --loglevel error --prefix "/opt/iobroker" (System call)
upload [3] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/words.js words.js application/javascript
Update "system.adapter.admin.0"
upload [2] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index_m.html index_m.html text/html
upload [1] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index.html index.html text/html
upload [0] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/admin.png admin.png image/png
Adapter "admin" updated
```

Still no change (web interface shows old version installed). So I hit the 'upgrade' button the web interface and it shows this:

```
$ ./iobroker upgrade admin
Update admin from @3.6.12 to @4.0.8
NPM version: 6.13.4
npm install iobroker.admin@4.0.8 --loglevel error --prefix "/opt/iobroker" (System call)
upload [3] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/words.js words.js application/javascript
upload [2] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index_m.html index_m.html text/html
upload [1] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index.html index.html text/html
upload [0] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/admin.png admin.png image/png
Adapter "admin" updated
process exited with code 0
```

After this the Web interface shows the correct Admin version installed. But you see the paths are identical in both the updates. I look in that path and there is only one 'node_modules' directory with the correct modules beneath.

Moreover, at the top right of the web interface there is a version number shown:

![image](https://user-images.githubusercontent.com/3794098/79681478-db67e080-821a-11ea-8683-f405b5238bf0.png)

This version number has always agreed with that returned by the command line query.

So I look at another module with a discrepancy: discovery.

```
# iobroker update | grep discovery
Adapter    "discovery"     : 2.3.2    , installed 2.3.1 [Updateable]
# cat /opt/iobroker/node_modules/iobroker.discovery/package.json
{
  "_from": "iobroker.discovery@2.3.1",
... snip
```

But the web interface (now running Admin 4.0.8) shows:

![image](https://user-images.githubusercontent.com/3794098/79681420-7d3afd80-821a-11ea-8edb-8f8e9541df3b.png)

Where does this web interface pull those versions from? Does it have it's own storage for versions that is only updated when that update is run from the actual web interface?

**Comment by Apollon77** [comment]
Hm ... do you have Single Host or multi host System?

Adapter versions are displayed in admin depending on the selected host on top.

**Comment by raintonr** [comment]
Ahhhhhhh... that is the problem!

I did try multi-host a while ago but found it problematic. I had to basically re-install everything on the master. Somehow this host (which was the 'master') still has remnants of the other host.

So this is what happened:

- I installed on this host `pl.mh`.
- I tried to add a second slave `pg.mh`. This didn't work as I expected so removed it. At least I thought I had removed it but seems not!
- The other host `pg.mh` is incorrectly selected at the top of the screen (look at the original screenshot I include). When I change this to `pl.mh` everything is correct.

So how do I completely remove all traces of that other host correctly?

In a second error, seems I didn't disable iobroker controller on the `pg.mh` host and it got restarted on system boot. If I had done this correctly would have got the 'Host is offline' message and noticed this sooner.

Thank you all for your help. This was slightly embarrassing :(
